src/main/java/com/fakng/fakngagrgtr/parser/stripe/StripeParser.java

```diff
@@ -0,0 +1,150 @@
+                        @Value("${url.stripe}") String url) {
+        super(htmlWebClient, companyRepository, locationProcessor);
+        this.url = url;
+    }
+
+    @PostConstruct
+    public void init() {
+        initBase();
+    }
+
+    @Override
+    protected String getCompanyName() {
+        return "Stripe";
+    }
+    @Override
+    public List<Vacancy> getAllVacancies() throws IOException {
+        HtmlPage page = downloadPage(String.format(url, 0));
+        int pageCount = getPageCount(page);
+        List<Vacancy> vacancyList = new ArrayList<>(fillVacancies(
+                page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+        ));
+
+        for (int i = 1; i <= pageCount; i++) {
+            page = downloadPage(String.format(url, i));
+            vacancyList.addAll(fillVacancies(
+                    page.getByXPath(VACANCIES_URLS_XPATH), page.getByXPath(VACANCY_CITIES_XPATH)
+            ));
+        }
+
+        return vacancyList;
+    }
+
+    private int getPageCount(HtmlPage page) {
+        String url = ((DomElement) page.getFirstByXPath(JOBS_PAGINATION_LIST_XPATH)).getAttribute("href");
+        return Integer.parseInt(url.replace("?skip=", "")) / PAGE_SIZE;
+    }
+
+    private List<Vacancy> fillVacancies(List<DomNode> urls, List<DomElement> cities) throws IOException {
+        List<Vacancy> vacancies = new ArrayList<>();
+
+        for (int i = 0; i < urls.size(); i++) {
+            String jobId = urls.get(i).getAttributes().getNamedItem("href").getNodeValue();
+
+            // Skip repeating vacancies and parse locations for them
+            List<LocationDTO> locationDTOs = new ArrayList<>();
+            while (jobId.equals(urls.get(i).getAttributes().getNamedItem("href").getNodeValue())) {
+                locationDTOs.add(createLocationDto(cities.get(i)));
+                i++;
+                if (i >= urls.size()) break;
+            }
+
+            // Variable i stopped in incorrect position
+            i--;
+            vacancies.add(createVacancy(jobId, locationDTOs));
+        }
+
+        return vacancies;
+    }
+
+    private Vacancy createVacancy(String jobId, List<LocationDTO> locationDTOS) throws IOException {
+        Vacancy vacancy = new Vacancy();
+
+        HtmlPage page = downloadPage(STRIPE_URL + jobId);
+        DomElement title = page.getFirstByXPath(VACANCY_TITLE_XPATH);
+        DomElement jobType = ((DomElement) page.getFirstByXPath(VACANCY_JOB_TYPE_XPATH)).getNextElementSibling();
+        DomElement team = ((DomElement) page.getFirstByXPath(VACANCY_TEAM_XPATH)).getNextElementSibling();
+
+        vacancy.setTitle(title.getTextContent());
+        vacancy.setDescription(createFullDescription(
+                createDescription(page), jobType.getTextContent(), team.getTextContent())
+        );
+        vacancy.setUrl(STRIPE_URL + jobId);
+        vacancy.setJobId(jobId.replaceAll("/.*/", ""));
```

Comment:
Okay, let's put it this way:
you have something like this: https://stripe.com/something/another-something/1234566
How to get the last number? Do you really need all your regex (which is slow) to find this number, when you know for sure that it will be the last piece of data after the LAST slash in this string?

Regex is not readable by anyone, hard to maintain, slow and tricky to deal with when there are edge cases. Please, write the code as simply as possible. Usually, this is the right way